src/nextcloudUtils.ts

```diff
@@ -0,0 +1,119 @@
+import { Milestone } from '@octokit/webhooks-types'
+import { LABEL_BACKPORT, LABEL_TO_REVIEW, LEARN_MORE, STEP_AMEND_SKIP_CI, STEP_REMOVE_EMPTY_COMMITS, STEP_REVIEW_CONFLICTS, Task, WARN_CONFLICTS, WARN_DIFF } from './constants'
+
+const compareSemanticVersions = (a: string, b: string) => {
+	// 1. Split the strings into their parts.
+	const a1 = a.split('.')
+	const b1 = b.split('.')
+
+	// 2. Contingency in case there's a 4th or 5th version
+	const len = Math.min(a1.length, b1.length)
+
+	// 3. Look through each version number and compare.
+	for (let i = 0; i < len; i++) {
+		const a2 = +a1[ i ] || 0
+		const b2 = +b1[ i ] || 0
+		
+		if (a2 !== b2) {
+			return a2 > b2 ? 1 : -1
+		}
+	}
+	
+	// 4. We hit this if the all checked versions so far are equal
+	return b1.length - a1.length
+}
+
+export const getMilestoneFromBase = (branch: string, milestones: Milestone[]): Milestone => {
+	// Extract the version from the branch name, e.g. stable21
+	const version = branch.match(/stable(\d{2})/i)?.[1]
+	if (!version) {
+		throw new Error(`Could not extract version from branch ${branch}`)
+	}
+	const selection = milestones
+		.filter(milestone => milestone.title.startsWith('Nextcloud ' + version))
```

Comment:
Could we return to https://github.com/nextcloud/backportbot/pull/388/files#diff-b7b030aed4ec199a8bc4801602fe6c4a72868e111f692b08d08278bb1946ce3cL15-L16
So it works for Talk's milestones again?
https://github.com/nextcloud/spreed/milestones

```suggestion
		.filter(milestone => milestone.title.contains(version))
```

That would work a bit better with "decoupled" apps that have other majors/minors than server.

Reply:
I'll use the old way, sure :)

Reply:
Done